pkg/operator/provisioner.go

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Some of the code below came from https://github.com/coreos/etcd-operator
+which also has the apache 2.0 license.
+*/
+package operator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-incubator/external-storage/lib/controller"
+	"github.com/rook/rook/pkg/model"
+	rookclient "github.com/rook/rook/pkg/rook/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type rookVolumeProvisioner struct {
+	clusterManager *clusterManager
+
+	// Configuration of rook volume provisioner
+	provConfig provisionerConfig
+}
+
+type provisionerConfig struct {
+	// Required:  the pool name to provision volumes from.
+	pool string
+
+	// Optional: Namespace of the cluster. Default is `rook`
+	clusterNamespace string
+
+	// Optional: Name of the cluster. Default is `rook`
+	clusterName string
+}
+
+func newRookVolumeProvisioner(clusterManager *clusterManager) controller.Provisioner {
+	return &rookVolumeProvisioner{
+		clusterManager: clusterManager,
+	}
+}
+
+// Provision creates a storage asset and returns a PV object representing it.
+func (p *rookVolumeProvisioner) Provision(options controller.VolumeOptions) (*v1.PersistentVolume, error) {
+
+	var err error
+	if options.PVC.Spec.Selector != nil {
+		return nil, fmt.Errorf("claim Selector is not supported")
+	}
+
+	glog.V(4).Infof("rookVolumeProvisioner: VolumeOptions %v", options)
+
+	cfg, err := parseClassParameters(options.Parameters)
+	if err != nil {
+		return nil, err
+	}
+	p.provConfig = *cfg
+
+	glog.V(4).Infof("rookVolumeProvisioner: creating volume with configuration %+v", p.provConfig)
+
+	capacity := options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
+	requestBytes := capacity.Value()
+
+	imageName := fmt.Sprintf("kubernetes-dynamic-%s-%s", options.PVName, uuid.NewUUID())
+
+	rookClient, err := p.clusterManager.getRookClient(p.provConfig.clusterNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get rook client: %v", err)
+	}
+
+	res, err := createVolume(imageName, p.provConfig.pool, requestBytes, rookClient)
+	if err != nil {
+		return nil, err
+	}
+	glog.V(1).Infof("Rook block image created: %s", res)
```

Comment:
We started using capnslog because most CoreOS projects we were using/working with were using that logging system.  It has been working well for us I'd say, so I don't see an urgent need to switch right now.

Note that using capnslog would remove the need to prefix your log statements with `rookVolumeProvisioner:` (if this provisioner is in a separate package that is)

We could consider switching wholesale, in a separate issue, so we do not have 2 logging frameworks in our dependencies.  Note that would only work if none of our other dependencies use capnslog, which I think is not the case.